cfpb/cfgov-refresh
# HOLD: Demo alternate sharing approach

**Do not merge!** This PR is intended to demonstrate a prototype of [this](https://github.com/cfpb/wagtail-sharing) alternate sharing approach. 

## Additions

- Adds dependence on [cfpb/wagtail-sharing](https://github.com/cfpb/wagtail-sharing) app for sharing of draft Wagtail content.

## Changes

- Use `wagtailsharing` URL router instead of Wagtail default.
- Disable reliance on a staging server hostname (e.g. `content.localhost:8000`) in favor of a hostname that allows for viewing of latest draft content.

## Testing

- Set the `WAGTAILSHARING_HOSTNAME` environment variable before you run your local server:

 ```sh
$ export WAGTAILSHARING_HOSTNAME='content.localhost:8000'
$ ./runserver.sh
 ```

- Run your local server. Create a new (draft only) page, and then also modify an existing published page with an unpublished change.
- Browse to these pages using the standard [http://localhost:8000](http://localhost:8000) hostname. You should get a 404 (not published yet) and see the old content, respectively.
- Browse to these pages using the new [http://content.localhost:8000](http://content.localhost:8000) hostname. You should see the draft content with a warning banner.

## Screenshots

![image](https://cloud.githubusercontent.com/assets/654645/22665873/6a4517bc-ec84-11e6-947d-ef1c2d69b43e.png)

## Checklist
